config/routes.rb

```diff
@@ -0,0 +1,12 @@
+Rails.application.routes.draw do
+  devise_for :users
+  # resources :categories
+  # resources :deals
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root "categories#index"
+  resources :users
+  resources :categories, only: [:index, :show, :new, :create, :destroy] do
+    resources :deals, only: [:index, :show, :new, :create, :destroy]
+  end
```

Comment:
- [ ] Kindly ensure you use only the necessary routes for your app eg you don't have a delete functionality yet you have a route for it, kindly fix it.